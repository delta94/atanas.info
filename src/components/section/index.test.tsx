import * as React from 'react';
import { shallow } from 'enzyme';

import { Section, SectionElements } from '.';

describe('Section component', () => {
	it('Should render the Section component', () => {
		const wrapper = shallow(
			<Section id="section" className=" test" hasButton={true} hasShell={true}>
				<p>Test</p>
				<p>Test</p>
			</Section>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('Should render the Section component without a button', () => {
		const wrapper = shallow(
			<Section id="section" hasButton={false} hasShell={true} shellClass="o-shell--flex">
				<p>Test</p>
				<p>Test</p>
			</Section>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('Should render the Section component without a shell', () => {
		const wrapper = shallow(
			<Section id="section" className=" test" hasButton={true} hasShell={false}>
				<p>Test</p>
				<p>Test</p>
			</Section>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('Should render the Section component without a button and a shell', () => {
		const wrapper = shallow(
			<Section id="section" className=" test" hasButton={false} hasShell={false}>
				<p>Test</p>
				<p>Test</p>
			</Section>
		);

		expect(wrapper).toMatchSnapshot();
	});
});

describe('SectionElement component', () => {
	it('Should render the SectionElements component', () => {
		const wrapper = shallow(
			<SectionElements id="section" className=" test" hasButton={true} hasShell={true}>
				<p>Test</p>
				<p>Test</p>
			</SectionElements>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('Should render the SectionElements component without a button', () => {
		const wrapper = shallow(
			<SectionElements id="section" hasButton={false} hasShell={true} shellClass="o-shell--flex">
				<p>Test</p>
				<p>Test</p>
			</SectionElements>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('Should render the SectionElements component without a shell', () => {
		const wrapper = shallow(
			<SectionElements id="section" className=" test" hasButton={true} hasShell={false}>
				<p>Test</p>
				<p>Test</p>
			</SectionElements>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('Should render the SectionElements component without a button and a shell', () => {
		const wrapper = shallow(
			<Section id="section" className=" test" hasButton={false} hasShell={false}>
				<p>Test</p>
				<p>Test</p>
			</Section>
		);

		expect(wrapper).toMatchSnapshot();
	});
});
