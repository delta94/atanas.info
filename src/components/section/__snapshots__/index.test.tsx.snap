// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component Should render the Section component 1`] = `
<section
  className="c-section c-section--section test"
  id="section"
>
  <div
    className="o-shell"
  >
    <Component
      className=" test"
      hasButton={true}
      hasShell={true}
      id="section"
    >
      <p>
        Test
      </p>
      <p>
        Test
      </p>
    </Component>
  </div>
</section>
`;

exports[`Section component Should render the Section component without a button 1`] = `
<section
  className="c-section c-section--section"
  id="section"
>
  <div
    className="o-shell o-shell--flex"
  >
    <Component
      hasButton={false}
      hasShell={true}
      id="section"
      shellClass="o-shell--flex"
    >
      <p>
        Test
      </p>
      <p>
        Test
      </p>
    </Component>
  </div>
</section>
`;

exports[`Section component Should render the Section component without a button and a shell 1`] = `
<section
  className="c-section c-section--section test"
  id="section"
>
  <Component
    className=" test"
    hasButton={false}
    hasShell={false}
    id="section"
  >
    <p>
      Test
    </p>
    <p>
      Test
    </p>
  </Component>
</section>
`;

exports[`Section component Should render the Section component without a shell 1`] = `
<section
  className="c-section c-section--section test"
  id="section"
>
  <Component
    className=" test"
    hasButton={true}
    hasShell={false}
    id="section"
  >
    <p>
      Test
    </p>
    <p>
      Test
    </p>
  </Component>
</section>
`;

exports[`SectionElement component Should render the SectionElements component 1`] = `
<Fragment>
  <p>
    Test
  </p>
  <p>
    Test
  </p>
  <div
    className="c-section__actions"
  >
    <Component
      className="c-btn"
      href="mailto:hi@atanas.info"
    >
      Hire me
    </Component>
  </div>
</Fragment>
`;

exports[`SectionElement component Should render the SectionElements component without a button 1`] = `
<Fragment>
  <p>
    Test
  </p>
  <p>
    Test
  </p>
</Fragment>
`;

exports[`SectionElement component Should render the SectionElements component without a button and a shell 1`] = `
<section
  className="c-section c-section--section test"
  id="section"
>
  <Component
    className=" test"
    hasButton={false}
    hasShell={false}
    id="section"
  >
    <p>
      Test
    </p>
    <p>
      Test
    </p>
  </Component>
</section>
`;

exports[`SectionElement component Should render the SectionElements component without a shell 1`] = `
<Fragment>
  <p>
    Test
  </p>
  <p>
    Test
  </p>
  <div
    className="c-section__actions"
  >
    <Component
      className="c-btn"
      href="mailto:hi@atanas.info"
    >
      Hire me
    </Component>
  </div>
</Fragment>
`;
